examples/dual_variable_example/dual_variable_example.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+
+"""
+General description
+-------------------
+
+A basic example to show how to get the dual variables from the system. Try
+to understand the plot.
+
+
+Installation requirements
+-------------------------
+
+This example requires the version v0.5.x of oemof. Install by:
+
+    pip install 'oemof.solph>=0.5,<0.6'
+
+Optional:
+
+    pip install matplotlib
+
+SPDX-FileCopyrightText: Uwe Krien <krien@uni-bremen.de>
+
+SPDX-License-Identifier: MIT
+"""
+
+
+# ****************************************************************************
+# ********** PART 1 - Define and optimise the energy system ******************
+# ****************************************************************************
+
+###############################################################################
+# imports
+###############################################################################
+
+import logging
+import os
+import pprint as pp
+
+import pandas as pd
+from oemof.solph import EnergySystem, Model, buses
+from oemof.solph import components as cmp
+from oemof.solph import flows, processing, views
+from oemof.tools import logger
+
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    plt = None
+
+
+solver = "cbc"  # 'glpk', 'gurobi',....
+debug = False  # Set number_of_timesteps to 3 to get a readable lp-file.
+number_of_time_steps = 48
+solver_verbose = False  # show/hide solver output
+
+# initiate the logger (see the API docs for more information)
+logger.define_logging()
+
+date_time_index = pd.date_range(
+    "1/1/2012", periods=number_of_time_steps, freq="H"
+)
+
+energysystem = EnergySystem(timeindex=date_time_index)
```

Comment:
```suggestion
energysystem = EnergySystem(timeindex=date_time_index, infer_last_interval=True)
```
We should make sure the new default behaviour is adhered to.

Reply:
:+1: